packages/cli/src/oclif/commands/invoke.js

```diff
@@ -300,7 +312,261 @@ const customLogger = (message, data) => {
+        throw new Error(
+          'CLIENT_ID and CLIENT_SECRET must be set in the .env file in non-interactive mode.'
+        );
+      } else {
+        console.warn(
+          'CLIENT_ID and CLIENT_SECRET are required for OAuth2, ' +
+            "but they are not found in the .env file. I'll prompt you for them now."
+        );
+      }
+    }
+
+    if (!process.env.CLIENT_ID) {
+      env.CLIENT_ID = await this.prompt('CLIENT_ID:', { useStderr: true });
+      process.env.CLIENT_ID = env.CLIENT_ID;
+    }
+    if (!process.env.CLIENT_SECRET) {
+      env.CLIENT_SECRET = await this.prompt('CLIENT_SECRET:', {
+        useStderr: true,
+      });
+      process.env.CLIENT_SECRET = env.CLIENT_SECRET;
+    }
+
+    if (!_.isEmpty(env)) {
+      // process.env changed, so we need to reload the modules that have loaded
+      // the old values of process.env
+      getLocalAppHandler({ reload: true });
+
+      // Save envs so the user won't have to re-enter them if the command fails
+      await appendEnv(env);
+      console.warn('CLIENT_ID and CLIENT_SECRET saved to .env file.');
+    }
+
+    startSpinner('Invoking authentication.oauth2Config.authorizeUrl');
+
+    const stateParam = crypto.randomBytes(20).toString('hex');
+    let authorizeUrl = await localAppCommand({
+      command: 'execute',
+      method: 'authentication.oauth2Config.authorizeUrl',
+      bundle: {
+        inputData: {
+          response_type: 'code',
+          redirect_uri: redirectUri,
+          state: stateParam,
+        },
+      },
+    });
+    if (!authorizeUrl.includes('&scope=')) {
+      const scope = appDefinition.authentication.oauth2Config.scope;
+      if (scope) {
+        authorizeUrl += `&scope=${encodeURIComponent(scope)}`;
+      }
+    }
+    debug('authorizeUrl:', authorizeUrl);
+
+    endSpinner();
+    startSpinner('Starting local HTTP server');
+
+    let resolveCode;
+    const codePromise = new Promise((resolve) => {
+      resolveCode = resolve;
+    });
+
+    const server = http.createServer((req, res) => {
+      // Parse the request URL to extract the query parameters
+      const code = new URL(req.url, redirectUri).searchParams.get('code');
+      if (code) {
+        resolveCode(code);
+        debug(`Received code '${code}' from ${req.headers.referer}`);
+
+        res.writeHead(200, { 'Content-Type': 'text/plain' });
+        res.end(
+          'Parameter `code` received successfully. Go back to the terminal to continue.'
+        );
+      } else {
+        res.writeHead(400, { 'Content-Type': 'text/plain' });
+        res.end(
+          'Error: Did not receive `code` query parameter. ' +
+            'Did you have the right CLIENT_ID and CLIENT_SECRET? ' +
+            'Or did your server respond properly?'
+        );
+      }
+    });
+
+    await new Promise((resolve) => {
+      server.listen(port, resolve);
+    });
+
+    endSpinner();
+    startSpinner('Opening browser to authorize');
```

Comment:
I got stuck at authorizing bc of bad params, I reckon it's hard to kill this process if the browser request errors?
Do you think it's valuable to set a timer for the code promise or provide 'ctrl/cmd + c' to abort helper?

Reply:
Yes it does, thank you for adding the hint text